### PR TITLE
feat(onboard/newpattern): suggest and submit a new pattern

### PR DESCRIPTION
**Pull request by tiloyi**
## I have read [the contributing guidelines](https://mozaic.adeo.cloud/contributing)

- [x] Yes
- [ ] No

## Does this PR introduce a breaking change?

read [What is a breaking change ?](/Contributing/GitConventions/)

- [ ] Yes
- [x] No

## Describe the changes

Add a documentation to suggest and submit a new pattern

Github issue number or Jira issue URL: **MZC-149**

### CODE REVIEW

File: src/data/icons.json [hunk 1]
```diff
@@ -122,38 +122,38 @@
   {
```
Comment (gael-boyenval) [inline src/data/icons.json]: remove this file form history

File: src/docs/Contributing/NewPattern/SubmitNewPattern/index.mdx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+Plop is a simple generator.  
+We use it to save bootstrapping time when creating new pattern.
+
+To use it, simply type in your terminal `npx plop pattern`
+
+Then the CLI will ask you 3 questions : 
+
+- the pattern name : use a Capital first letter and no spaces (ex: Buttons)
+- the pattern path : from the `src/docs` directory with no `/` at the end (ex: Components)
```
Comment (gael-boyenval) [inline src/docs/Contributing/NewPattern/SubmitNewPattern/index.mdx]: maybe something need to be changed into the plop js file to set path to src docs ?

File: src/docs/Contributing/NewPattern/SubmitNewPattern/index.mdx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+- the pattern order : require an integer for the main menu ex: 4
```
Comment (gael-boyenval) [inline src/docs/Contributing/NewPattern/SubmitNewPattern/index.mdx]: maybe we need to be more specific about order, it's not very clear

File: src/docs/Contributing/NewPattern/SubmitNewPattern/index.mdx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+At the end of the process Plop will generate the required `usage`, `code`, `design`, `specs` tabs with helping markdown content.
+
+Plop will also generate a basic preview.
```
Comment (gael-boyenval) [inline src/docs/Contributing/NewPattern/SubmitNewPattern/index.mdx]: maybe we should detail more what should be found in the 4 tabs.

File: src/docs/Contributing/NewPattern/AcceptanceCriterias/index.mdx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+The code need to be reviewed by the core team front-end. 
+
+**The core team front-end need to assess :**
+
+* the code standards, naming convention, and code comments
+* the consistency related to usage, configuration...
+* the overall quality regarding accessibility, SEO, compatibility etc
+* the retro-compatibility and possible collisions with other patterns and the rest of the code source
+* the correct tagging, labelling and categorisation of the pattern and our git flow implementation
+
```
Comment (gael-boyenval) [inline src/docs/Contributing/NewPattern/AcceptanceCriterias/index.mdx]: * the quality of the code examples
* that enough informations are provided about accessibility and html semantic (what tags to use, aria attributes etc)

File: src/data/icons.json (hunk 1 above)
Comment (tiloyi) [reply src/data/icons.json]: fixed

File: src/data/icons.json [hunk 6]
```diff
@@ -1,2486 +0,0 @@
-[
```
[the remark this answers is not in the document]
Comment (tiloyi) [reply src/data/icons.json]: fixed

File: src/docs/Contributing/NewPattern/SubmitNewPattern/index.mdx [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+To use it, simply type in your terminal `npx plop pattern`
+
+Then the CLI will ask you 3 questions : 
+
+- the pattern name : use a Capital first letter and no spaces (ex: Buttons)
+- the pattern path : from the `src/docs` directory with no `/` at the end (ex: Components)
+- the pattern order : the position of your pattern in the navigation menu (an integer - ex: 4)
+
+At the end of the process Plop will generate the required `usage`, `code`, `design`, `specs` tabs with helping markdown content.
```
Comment (caqueste) [inline src/docs/Contributing/NewPattern/SubmitNewPattern/index.mdx]: You must reverse design and specs, to have the same order that we have on the site : 1. usage, 2. code, 3. specs and 4. design

Comment (tiloyi) [reply src/docs/Contributing/NewPattern/SubmitNewPattern/index.mdx]: It's fixed and I added a link at the top of the page to redirect designers to definition of done.

**Comment by gael-boyenval** [comment]
@caqueste @caqueste @caqueste ? can you review the changes you asked for ?

**Pull request merged**